pygdf/cudautils.py

```diff
@@ -377,6 +377,36 @@ def compute_scale(arr, vmin, vmax):
     return out
 
 
+@cuda.jit
+def gpu_label(arr, cat, val, out):
+    i = cuda.grid(1)
+    if i < out.size:
+        res = (val) if (arr[i] == cat) else arr[i]
+        out[i] = res
+
+
+def apply_label(arr, cat, val, dtype):
+    """
+
+    Parameters
+    ----------
+    arr : device array
+        data
+    cat : Unique category value
+    val : scalar
+        value to compare against
+    dtype : np.dtype
+        output array dtype
+
+    Returns
+    -------
+    result : device array
+    """
+    out = cuda.device_array(shape=arr.size, dtype=dtype)
```

Comment:
Beware that `device_array` doesn't initialize the allocation at all.  The `out` array may contain random values.  Better initialize it.

Reply:
I did get what you mean by that, doesn't `out` array will by default be having all values as `0` ?

Reply:
No, it doesn't get initialized at all.  It just calls `cuMemAlloc` under the hood.

Reply:
In this case, isn't it fine if the array is initialized? since values are being added from `arr` in `gpu_label`